# Phragmites-australis-transcriptome-optimal-assembly
Workflow and scripts used for the optimal assembly of the *Phragmites australis* transcriptome.

## Purpose
Multiple types of tissue were used to construct forty-nine *P. australis* transcriptomes via different assembly tools and multiple parameter settings. The optimal transcriptome for functional annotation and downstream analyses was selected among these transcriptomes by comprehensive assessments. For a total of 422,589 transcripts assembled in this transcriptome, 319,046 transcripts (75.5%) have at least one functional annotation. Within the transcriptome, 1,495 transcripts showing tissue-specific expression pattern, 10,828 putative transcription factors, and 72,165 simple sequence repeats markers were further identified. With this optimal transcriptome and all relative information from downstream analyses, foundations for future studies on the mechanisms underlying the invasiveness of non-native *P. australis* were laid.

## Data
1. *Phragmites australis* four tissues: leaf, shoot meristem, rhizome and inflorescence.

    Illumina paired end rna_seq data, 100bp, detailed information and trimmed information could be checked below
2. *Setaria italica* v2.2(https://data.jgi.doe.gov/refine-download/phytozome/cladeId:416/All/proteomeId:312/list)
3. *Phragmites australis* leaf culm and rhizome tip tissue transcriptome sequence reads from PRJNA314710 (https://www.ncbi.nlm.nih.gov/bioproject/PRJNA314710)
4. BUSCO library: poales_odb10 (https://busco-data.ezlab.org/v4/data/lineages/poales_odb10.2019-11-20.tar.gz) and embryophyta_odb10 (https://busco-data.ezlab.org/v4/data/lineages/embryophyta_odb10.2019-11-20.tar.gz)
5. Uniprot Swiss-Prot (ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz)
6. Pfam (ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz)
7. 
8.


## Software






